virttest/qemu_vm.py

```diff
@@ -2342,6 +2338,22 @@ def sort_key(dev):
                 if ats:
                     add_virtio_option("ats", ats, devices, device, dev_type)
 
+        # Add extra root_port at the end of the command line only if there is
+        # free slot on pci.0, discarding them otherwise
+        from virttest.qemu_devices import utils
```

Comment:
imports should be on top of the file.

Reply:
also as there is just the DeviceError being used (and given to generic name) I'd suggest:

```diff
diff --git a/virttest/qemu_vm.py b/virttest/qemu_vm.py
index 0dc1798c..4d0fef26 100644
--- a/virttest/qemu_vm.py
+++ b/virttest/qemu_vm.py
@@ -35,6 +35,7 @@ from virttest import storage
 from virttest import error_context
 from virttest.compat_52lts import decode_to_text
 from virttest.qemu_devices import qdevices, qcontainer
+from virttest.qemu_devices.utils import DeviceError
 
 
 class QemuSegFaultError(virt_vm.VMError):
@@ -2340,7 +2341,6 @@ class VM(virt_vm.BaseVM):
 
         # Add extra root_port at the end of the command line only if there is
         # free slot on pci.0, discarding them otherwise
-        from virttest.qemu_devices import utils
         pcie_extra_root_port = int(params.get('pcie_extra_root_port', 0))
         for num in range(pcie_extra_root_port):
             try:
@@ -2348,7 +2348,7 @@ class VM(virt_vm.BaseVM):
                     "pcie_extra_root_port_%s"
                     % num, {"type": "pcie-root-port"})
                 devices.insert(dev)
-            except utils.DeviceError:
+            except DeviceError:
                 logging.warning("No sufficient free slot for extra"
                                 " root port, discarding %d of them"
                                 % (pcie_extra_root_port - num))
```